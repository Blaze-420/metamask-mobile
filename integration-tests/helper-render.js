import React from 'react';
import { render } from '@testing-library/react';
import { ConnectedRoot } from '../app/components/Views/Root/ConnectedRoot';
import { createStoreAndPersistor } from '../app/store';

export async function renderIntegrationTest({ preloadedState, ...renderOptions }) {
  const { store, persistor } = await createStoreAndPersistor(preloadedState).catch(console.error);

  return renderWithRedux({ store, persistor, renderOptions});
}

function renderWithRedux({ store, persistor, renderOptions }) {
  return render(
    <ConnectedRoot store={store} persistor={persistor} />,
    renderOptions,
  );
}
