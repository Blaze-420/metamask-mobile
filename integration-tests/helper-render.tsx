import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ConnectedRoot} from '../app/components/Views/Root/ConnectedRoot';
import { createStoreAndPersistor } from '../app/store';
import { RootState } from '../app/reducers';

export async function renderIntegrationTest({
  preloadedState,
  renderOptions
}: {
  preloadedState: RootState,
  renderOptions: RenderOptions
}) {
  try {
    const { store, persistor } = await createStoreAndPersistor(preloadedState);
    await new Promise((resolve) => setTimeout(resolve, 1000));
    return {
      ...render(
        <ConnectedRoot store={store} persistor={persistor} />,
        { ...renderOptions },
      )
    };
  } catch (error) {
    console.error('error', error);
    throw error;
  }
}
