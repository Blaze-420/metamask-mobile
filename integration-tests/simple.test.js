import { act, screen } from '@testing-library/react-native';
import FixtureBuilder from '../e2e/fixtures/fixture-builder';
import { renderIntegrationTest } from './helper-render';

describe('Simple test', () => {
  it('should be true', async () => {
    const preloadedState = new FixtureBuilder().withDefaultFixture().build().state;
    await act(async () => {
      await renderIntegrationTest({ preloadedState });
    });
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
