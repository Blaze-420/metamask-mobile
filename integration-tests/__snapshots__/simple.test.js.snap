// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple test should be true 1`] = `null`;
